src/test/ui/nll/issue-55651.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// run-pass
```

Comment:
Hmm. I know I asked for this to be a `// run-pass` test, but I was at the time thinking of something that would actually do some sort of runtime validation of the resulting state...

I cannot off the top of my head come up with an obviously good change to this to make it validate its state at runtime.

So lets just make this a `// compile-pass` test instead. That will accomplish my main goal of insuring that this code will continue to compile regardess of how the diagnostic output changes.

Reply:
Fixed.